Update default documentation based on #500 (#504)
astronomer/dag-factory: docs/configuration/defaults.md

# Defaults

DAG Factory allows you to define default values for DAG-level arguments and Airflow
[default_args](https://airflow.apache.org/docs/apache-airflow/stable/core-concepts/dags.html#default-arguments). There are several ways to accomplish this:

- Define the `default_args` within each DAG definition in the DAGs YAML file;
- Declare a `default` block within the toplevel of the DAGs YAML file;
- Define the `default_args_config_dict` argument when instantiating the `DAGFactory` class;
- Create one or multiple `defaults.yml` and declare the `default_args_config_path` argument in the `DAGFactory` class. This approach includes support for combining multiple `defaults.yml` files.

Although you cannot use the last two configurations together, you can use a combination of the first two configurations with either the third or the last.

Below, we detail how to use each of these approaches and also how to combine them.

## Specifying `default_args` directly in the DAG YAML specification

This configuration affects only the DAG where the `default_args` are defined.

> You can override or define specific `default_args` at the individual DAG level. This strategy allows you to customize arguments for each DAG without affecting others. Not only can existing `default_args` be overridden directly in a DAG configuration, but also adding new arguments.

Author: ```suggestion
You can override or define specific `default_args` at the individual DAG level. This strategy allows you to customize arguments for each DAG without affecting others. Not only can existing `default_args` be overridden directly in a DAG configuration, but also adding other DAG-level arguments.
```

> You can override or define specific `default_args` at the individual DAG level. This strategy allows you to customize arguments for each DAG without affecting others. Not only can existing `default_args` be overridden directly in a DAG configuration, but also adding other DAG-level arguments.

Reviewer: ```suggestion
You can override or define specific `default_args` at the individual DAG level. This strategy allows you to customize arguments for each DAG without affecting others. Not only can existing `default_args` be overridden directly in a DAG configuration, but other DAG-level arguments can be added.
```

> You can override or define specific `default_args` at the individual DAG level. This strategy allows you to customize arguments for each DAG without affecting others. Not only can existing `default_args` be overridden directly in a DAG configuration, but other DAG-level arguments can be added.

Reviewer: ```suggestion
You can override or define specific `default_args` at the individual DAG level. This strategy allows you to customize arguments for each DAG without affecting others. Not only can existing `default_args` be overridden directly in a DAG configuration, but also other DAG-level arguments can be added.
```

Sorry for another quick suggestion, we had the "also" there before, my previous suggestion removed that.

  ...
   ```

### Combining multiple `defaults.yml` files

It is possible to combine and merge the content of multiple `defaults.yml` files.

To accomplish this, you should declare in the `default_args_config_path` a folder that is a parent folder of a DAG-defined `YAML` file. In this case, DAG Factory will merge all the `defaults.yml` configurations, following the directories' hierarchy, and give precedence to the arguments declared in the `defaults.yml` file closest to the DAG YAML file.

As an example, let's say there are DagFactory DAGs defined inside the `a/b/c/some_dags.yml` file following this directory tree:

```shell
sample_project
└── a
    ├── b
    │   ├── c
    │   │   ├── defaults.yml
    │   │   └── some_dags.yml
    │   └── defaults.yml
    └── defaults.yml
```

Assuming you instantiate the DAG by using:

```python
DagFactory(
   "a/b/c/some_dags.yml",

>    default_args_config_path="a/b/c"

Author: ```suggestion
   default_args_config_path="a"
```
